src/Cake.NuGet/NuGetPackageInstaller.cs

```diff
@@ -175,6 +160,38 @@ public IReadOnlyCollection<IFile> Install(PackageReference package, PackageType
             return result;
         }
 
+        private static DirectoryPath GetPackagePath(DirectoryPath root, PackageReference package)
+        {
+            if (package.Parameters.ContainsKey("version"))
+            {
+                var version = package.Parameters["version"].First();
+                return root.Combine($"{package.Package}.{version}");
```

Comment:
Seeing issues because nuget supports partial nuget version numbers, getting this error building Cake with this PR
```
Installing NuGet package ReportGenerator...
Executing: /opt/nuget/nuget.exe install "ReportGenerator" -OutputDirectory "/pullrequest/tools" -Source "https://www.nuget.org/api/v2" -Version "2.4.5" -NonInteractive
Could not find any relevant files for tool 'ReportGenerator'. Perhaps you need an include parameter?
Error: Cake.Core.CakeException: Failed to install tool 'ReportGenerator'.
```
`ls tools` gives:
```
GitVersion.CommandLine.3.6.2
OpenCover.4.6.519
ReportGenerator.2.4.5.0
coveralls.io.1.3.4
gitreleasemanager.0.5.0 
```
Changing `#tool "nuget:https://www.nuget.org/api/v2?package=ReportGenerator&version=2.4.5"` to `#tool "nuget:https://www.nuget.org/api/v2?package=ReportGenerator&version=2.4.5.0"` in `build.cake` sorts the issue.

How smart do we need to be here? 

Checking nuspec of `ReportGenerator`
```xml
<version>2.4.5.0</version>
```
vs i.e. GitReleaseManager
```
<version>0.6.0</version>
```
but nuget omits the .0 in the UI so no good way of knowing that.

Perhaps `  return root.Combine($"{package.Package}.{version}");` should fetched with a globber and firstordefault best match.